### ISSUE TITLE
OnNodeHover not working once we drag the node outside container

### ISSUE THREAD

**Issue by vinodpotru** [vasturiano/react-force-graph#251]
**Issue:1 -> OnNodeHover not working once we drag the node  into outside container**
[https://vasturiano.github.io/react-force-graph/example/tree/](https://vasturiano.github.io/react-force-graph/example/tree/)
**_Note: I am Working on ForceGraph2D, version: "react-force-graph": "^1.37.1"_**
**_Steps:_**
-> Open above link and Observe as on node hover its shows title.
-> When we drag node into out side container(outside visible box) and then hover on any node but previous hover functionality is not working.

**Issue:2 -> Can't implement title on label text hover**
**_Steps:_**
-> I have created graph like node display as Image and label(shown below)
![image](https://user-images.githubusercontent.com/68293272/105334044-6e93d380-5bfc-11eb-8d4a-c121c292464a.png)

-> if label name is long then I have trimmed and show three dots
-> but I need to display full title if user hover on three dots
-> **Is this possible to show title on Label hover?**

**Issue:3 -> how to find event element in onNodeHover? like onNodeClick have second parameter is event.**
**_Scenario_**
-> I want to display customize popup when I hover on Node so I need offsetX, offsetY, pageX and pageY values of Node element. I can achieve this on OnNodeClick(node, event) but unable to find node events details in OnNodeHover(node, prevNode).

Is there aby possibility to get element on onNodeHover?
Or Any Way to find event element using node object index Id?

**Comment by vasturiano** [comment]
@vinodpotru thanks for reaching out.

Issue 1: Thanks for the report. This should now be fixed in `v1.37.2` of the main module.

Issue 2: Using `nodeLabel` you can set the content to display on node hover, such as your complete untrimmed label. You can also adjust the node hover radius area by setting the `nodeVal` per node.

Issue 3: The node objects include `x` and `y` attributes corresponding to their positions within the graph canvas. You can then convert those values to container coordinates using the utility function `graph2ScreenCoords(x, y)`.

**Comment by vinodpotru** [comment]
Thank you @vasturiano , Its working

**Comment by vasturiano** [comment]
@vinodpotru I don't believe it's possible to separate the hover functionalities of different parts of the node. There is a single hover area that is used to capture mouse events on.

The only way I can imagine this being possible is if you separate the image and the text into separate nodes. You would then have to keep their positions synchronized using `f*` attributes. It's not trivial, but possible if you need it.

**Comment by vinodpotru** [comment]
Thank you @vasturiano
